R/validate_inputs.R

```diff
@@ -116,14 +119,17 @@ do_validate_population <- function(population) {
 #'
 #' @return An error if invalid.
 #' @keywords internal
-do_validate_programme <- function(programme) {
-  programme <- read_csv(programme, header = TRUE)
-  assert_single_country(programme, "programme")
+do_validate_programme <- function(programme, shape) {
+  data <- read_csv(programme, header = TRUE)
+  assert_single_country(data, "programme")
   assert_column_names(
-    colnames(programme),
+    colnames(data),
     c("iso3", "area_id", "period", "sex", "age_group_id", "indicator", "value"))
-  list(data = programme,
-       filters = list("age" = get_age_filters(programme)))
+  assert_consistent_regions(read_regions(shape, "shape"),
+                            read_regions(programme, "programme"),
```

Comment:
One question is about this, we pass the path to the programme data into the function not the already read data. It means we're reading it twice, once on line 123 and once on line 129. How bad is that? Does R do something clever and not duplicate the file read or should I be handling that?

Reply:
I think it might be best to ignore this for now and we can look at general caching once we see the general lay of things